runtime/src/snapshot_utils.rs

```diff
@@ -932,46 +1551,86 @@ fn verify_snapshot_version_and_folder(
 }
 
 #[allow(clippy::too_many_arguments)]
-fn rebuild_bank_from_snapshots(
-    snapshot_version: &str,
+fn rebuild_bank_from_snapshots<P, Q>(
+    full_snapshot_version: &str,
+    incremental_snapshot_version: Option<&str>,
     frozen_account_pubkeys: &[Pubkey],
-    unpacked_snapshots_dir: &Path,
+    full_snapshot_unpacked_snapshots_dir: P,
+    incremental_snapshot_unpacked_snapshots_dir: Option<Q>,
     account_paths: &[PathBuf],
     unpacked_append_vec_map: UnpackedAppendVecMap,
     genesis_config: &GenesisConfig,
     debug_keys: Option<Arc<HashSet<Pubkey>>>,
     additional_builtins: Option<&Builtins>,
-    account_indexes: AccountSecondaryIndexes,
+    account_secondary_indexes: AccountSecondaryIndexes,
     accounts_db_caching_enabled: bool,
     limit_load_slot_count_from_snapshot: Option<usize>,
     shrink_ratio: AccountShrinkThreshold,
-) -> Result<Bank> {
-    let (snapshot_version_enum, root_paths) =
-        verify_snapshot_version_and_folder(snapshot_version, unpacked_snapshots_dir)?;
+) -> Result<Bank>
+where
+    P: AsRef<Path>,
+    Q: AsRef<Path>,
+{
+    let (full_snapshot_version_enum, full_snapshot_root_paths) =
+        verify_snapshot_version_and_folder(
+            full_snapshot_version,
+            &full_snapshot_unpacked_snapshots_dir,
+        )?;
+    let (incremental_snapshot_version_enum, incremental_snapshot_root_paths) = match (
+        incremental_snapshot_version,
+        &incremental_snapshot_unpacked_snapshots_dir,
+    ) {
+        (Some(snapshot_version), Some(unpacked_snapshots_dir)) => {
+            let (version, paths) =
+                verify_snapshot_version_and_folder(snapshot_version, unpacked_snapshots_dir)?;
+            (Some(version), Some(paths))
+        }
+        _ => (None, None),
+    };
     info!(
-        "Loading bank from {}",
-        &root_paths.snapshot_file_path.display()
+        "Loading bank from {} and {:?}",
```

Comment:
nit: `Loading bank from full snapshot: {} and incremental snapshot: {:?}`

Reply:
Same naming questions from here [1] on how to name snapshots.

[1] https://github.com/solana-labs/solana/pull/18504#discussion_r668915990

Reply:
Done!